plugins/plugins.gradle.kts

```diff
@@ -10,17 +10,18 @@ subprojects {
         }
         val sourceJar = task<Jar>("sourceJar") {
             description = "Creates a JAR that contains the source code."
-            from(java.sourceSets["main"].allSource)
+            from(project.sourceSets["main"].allSource)
             classifier = "sources"
         }
+        val javaPluginConvention = convention.plugins["java"] as JavaPluginConvention
         val javadocJar = task<Jar>("javadocJar") {
```

Comment:
Use `tasks.registering` now that that task providers should work as an argument to `artifact`.

Reply:
Nope.

---
Using `tasks.register`:

```
Cannot convert the provided notation to an object of type MavenArtifact: task ':plugins:base:javadocJar'.
The following types/formats are supported:
  - Instances of MavenArtifact.
  - Instances of AbstractArchiveTask, for example jar.
  - Instances of PublishArtifact
  - Maps containing a 'source' entry, for example [source: '/path/to/file', extension: 'zip'].
  - Anything that can be converted to a file, as per Project.file()
```

---

Using `tasks.registering`:
```
Cannot convert the provided notation to an object of type MavenArtifact: org.gradle.kotlin.dsl.RegisteringDomainObjectDelegateProviderWithTypeAndAction@7600c166.
The following types/formats are supported:
  - Instances of MavenArtifact.
  - Instances of AbstractArchiveTask, for example jar.
  - Instances of PublishArtifact
  - Maps containing a 'source' entry, for example [source: '/path/to/file', extension: 'zip'].
  - Anything that can be converted to a file, as per Project.file()
```

Reply:
Ah, I thought it had been fixed for 5.1. Aparently not.